e2e/cypress/integration/channel_sidebar/category_collapsing_unread_filter_spec.js

```diff
@@ -0,0 +1,294 @@
+
+        cy.visit('/');
+    });
+
+    it('should display collapsed state when collapsed', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team
+        cy.get('#headerTeamName').should('contain', teamName);
+
+        // * Verify that the category doesn't appear collapsed currently
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS) i').should('be.visible').should('not.have.class', 'icon-rotate-minus-90');
+
+        // # Click on PUBLIC CHANNELS
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS)').should('be.visible').click();
+
+        // * Verify that the category now appears collapsed
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS) i').should('have.class', 'icon-rotate-minus-90');
+    });
+
+    it('should collapse channels that are not the currently viewed channel', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team and are on Town Square
+        cy.get('#headerTeamName').should('contain', teamName);
+        cy.url().should('include', `/${teamName}/channels/town-square`);
+
+        // * Verify that both channels are visible when not collapsed
+        cy.get('.SidebarChannel:contains(Off-Topic)').should('be.visible');
+        cy.get('.SidebarChannel:contains(Town Square)').should('be.visible');
+
+        // # Click on PUBLIC CHANNELS
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS)').should('be.visible').click();
+
+        // * Verify that Off-Topic is no longer visible but Town Square still is
+        cy.get('.SidebarChannel:contains(Off-Topic)').should('not.be.visible');
+        cy.get('.SidebarChannel:contains(Town Square)').should('be.visible');
+    });
+
+    it('should collapse channels that are not unread channels', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team and are on Town Square
+        cy.get('#headerTeamName').should('contain', teamName);
+        cy.url().should('include', `/${teamName}/channels/town-square`);
+
+        // Create a new channel and post a message into it
+        cy.getCurrentTeamId().then((teamId) => {
+            cy.apiCreateChannel(teamId, 'channel-test', 'Channel Test').then((res) => {
+                cy.postMessageAs({sender: sysadmin, message: 'Test', channelId: res.body.id});
+
+                // * Verify that all channels are visible
+                cy.get('.SidebarChannel:contains(Off-Topic)').should('be.visible');
+                cy.get('.SidebarChannel:contains(Channel Test)').should('be.visible').should('has.class', 'unread');
```

Comment:
Test failed here. 
![Screen Shot 2020-03-20 at 7 31 26 AM](https://user-images.githubusercontent.com/5334504/77124201-30c69b80-6a7d-11ea-8279-d4e4ac1ad1ba.png)

"Channel Test" is not turning into unread since sysadmin is not channel member and is not able to successfully post a message.  Before `cy.postMessageAs` above, I suggests to explicitly add sysadmin:
```javascript
                const channel = res.body;
                cy.apiGetUserByEmail(sysadmin.email).then((emailResponse) => {
                    const user = emailResponse.body;
                    cy.apiAddUserToTeam(channel.team_id, user.id);
                    cy.apiAddUserToChannel(channel.id, user.id);
                });

                cy.postMessageAs({sender: sysadmin, message: 'Test', channelId: channel.id});
```

Reply:
Thanks for the tip, but this is actually causing the tests to fail consistently for me now. Do I need to wait for each API call maybe?

Reply:
Can you share the fail logs you're seeing?  May not need explicit wait since `cy.get` is doing it under the hood already.  How about try moving `cy.postMessageAs` and the rest of `cy.get` inside the last `.then` ==> 
```
cy.apiGetUserByEmail(sysadmin.email).then((emailResponse) => { // all here //}); 
```

Reply:
Still not working, it looks like the channel isn't being marked as unread using the above logic.
`CypressError: Timed out retrying: expected '<li.SidebarChannel>' to have class 'unread'`

Reply:
The systems admin shouldn't have to join the channel to post in it. System admins can do basically anything. This is probably failing because of something else.

Reply:
Yeah, one issue I'm seeing is on how Cypress handles websocket . Maybe instead of waiting for message via websocket, reload the page after posting a message.  Otherwise, I'm good and will take it from here as part of Cypress test maintenance.

Reply:
Okay, added the `reload()` after `postMessageAs()`. Let me know if that works for you.

Reply:
Thanks, works for me.